src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/SwapEndpointSpec.groovy

```diff
@@ -1373,6 +1376,66 @@ switches"() {
         !isSwitchValid && switches.each { northbound.synchronizeSwitch(it, true)}
     }
 
+    @Tidy
+    @Unroll
+    def "Able to swap endpoints for a flow with flowLoop "() {
+        given: "Two flows with the same source and destination switches"
+        flowHelper.addFlow(flow1)
+        flowHelper.addFlow(flow2)
+
+        and: "FlowLoop is created for the first flow on the dst switch"
+        northboundV2.createFlowLoop(flow1.id, new FlowLoopPayload(switchPair.dst.dpId))
+        Wrappers.wait(WAIT_OFFSET) {
+            assert northbound.getFlowStatus(flow1.id).status == FlowState.UP
+        }
+
+        when: "Try to swap endpoints for flows"
+        def response = northbound.swapFlowEndpoint(
+                new SwapFlowPayload(flow1.id, flowHelper.toFlowEndpointV2(flow1Src),
+                        flowHelper.toFlowEndpointV2(flow1Dst)),
+                new SwapFlowPayload(flow2.id, flowHelper.toFlowEndpointV2(flow2Src),
+                        flowHelper.toFlowEndpointV2(flow2Dst)))
+
+        then: "Endpoints are successfully swapped"
+        verifyEndpoints(response, flow1Src, flow1Dst, flow2Src, flow2Dst)
+        verifyEndpoints(flow1.id, flow2.id, flow1Src, flow1Dst, flow2Src, flow2Dst)
+
+        and: "FlowLoop is still created"
```

Comment:
I'm pretty sure you have no idea what have happened right now. Especially when you swapped `dst1<->dst2`. So the flow loop is still assigned to flow1, but did the system reinstall the rules according to the new dst2 endpoint? I have no idea, neither this test. This will be very visible if dst1 and dst2 are located on different switches. I have no certainty how swap will work in this case and what will happen with the loop and flows.

Reply:
refactored